lms/djangoapps/hera/fragments.py

```diff
@@ -213,41 +239,109 @@ class DashboardPageOutlineFragmentView(CourseOutlineFragmentView):
-                            popup = True
-                            break
+                if active_block_id == subsection['block_id'] and not subsection_active:
+                    course_block_id = subsection['block_id']
+                    subsection_active = subsection
+                if subsection['complete']:
+                    completed_subsections[subsection.get('block_id')] = {
+                        'display_name': subsection.get('display_name'),
+                        'lms_web_url': subsection.get('lms_web_url', '')
+                    }
 
-        units = incomplete_subsection.get('children', [{}])
-        start_over_url = units[0].get('lms_web_url', '')
+        for chapter in courseware_summary:
+            if not chapter['display_name'] == "hidden":
+                for subsection in chapter['sections']: # the sections in chapter are actually subsection
+                    if course_block_id == subsection.location.block_id:
+                        earned = subsection.all_total.earned
+                        total = subsection.all_total.possible
+                    if subsection.location.block_id in completed_subsections:
+                        try:
+                            points = int(subsection.all_total.earned / subsection.all_total.possible * 100)
+                        except ZeroDivisionError:
+                            points = 0
+                        completed_subsections[subsection.location.block_id].update({'points': points, 'medal': get_medal(points)})
 
-        xblock_display_names = self.create_xblock_id_and_name_dict(course_block_tree)
-        gated_content = self.get_content_milestones(request, course_key)
+        try:
+            earned = round(earned/total*10, 1)
+        except ZeroDivisionError:
+            earned = 0
+        selection_subsections, completed_subsection_ids = unit_grade_level(
+            earned,
+            course_block_tree,
+            subsection_active,
+            request,
+            course_id
+        )
+
+        ordered_subsections = []
+        popup = False
+        for subsection_id in completed_subsection_ids:
+            ordered_subsections.append(completed_subsections.get(subsection_id))
+        if not subsection_active['complete'] and len(ordered_subsections) < 8:
+            for unit in subsection_active.get('children', []):
+                if unit['complete']:
+                    popup = True
+                    ordered_subsections.append(subsection_active)
+                    break
+        if not popup:
+            for selection_subsection in selection_subsections['children']:
+                if not selection_subsection['complete'] and len(ordered_subsections) < 8:
+                    ordered_subsections.append(selection_subsection)
+        while len(ordered_subsections) < 8:
+            ordered_subsections.append(None)
+
+        try:
+            last_completed_subsection_id = completed_subsection_ids[-1]
```

Comment:
can we get. the last completed_subsection from the `ordered_subsections` above `While`?